### ISSUE TITLE
Keras callbacks?

### ISSUE THREAD

**Issue by kmader** [Losswise/losswise-python#1]
It would be great to have keras callbacks for using losswise rather than having to manually handle the graphs and sessions

**Comment by nicodjimenez** [comment]
For sure! I'm actually not super familiar with Keras callbacks.. Just to be clear, you want to be able to do something like this? 

```python
model.fit(x_train, y_train, batch_size=128, epochs=20, verbose=0, callbacks=[losswise_callback])
```

**Comment by kmader** [comment]
yea that would be great!

**Comment by nicodjimenez** [comment]
@kmader done! See https://docs.losswise.com/#keras-plugin for example usage. Let me know if you have any further issues.

**Comment by nicodjimenez** [comment]
Will keep this issue open until you can confirm the update solves this issue :)

**Comment by kmader** [comment]
Cool, thanks! I just noticed you have an argument use_acc to include accuracy. Given the number of different models (regression, segmentation, text generation...) that are commonly used, it would be really cool if you just tracked all of the metrics which were compiled with the model (```model.compile(..., metrics = ['acc', 'mae', ]```). That way you wouldn't have to additionally specify which metrics to track.

**Comment by nicodjimenez** [comment]
Keras callbacks only have access to the "logs" parameter: https://keras.io/callbacks/#create-a-callback

I thought that accuracy tracking was builtin thing.  If metrics = ['acc', 'mae', ] then will there for sure exist keys 'acc' and 'mae' in `logs`?

**Comment by kmader** [comment]
Exactly, so what you can do is iterate through the log dictionary to discover each field (with and without the 'val_' prefix) and then create a graph for each one (maybe a defaultdict) and then append the values. That way whatever metrics are included when the model is compiled are included in the losswise output (often we use custom functions or weighted functions and so having it automatically pick those up makes it nicer). The only mandatory element of logs is 'loss' everything else are only included if you specify them in metrics. Also if you do not provide a validation dataset, none of the 'val_' keys in logs are generated

**Comment by nicodjimenez** [comment]
Ok good call, just implemented this with 763649a448ad6ba79c1f0bd2013ab91f21c2654c.  Also added automatic inference of the max number of iterations.  So now you just need to do:

```python
model.fit(x_train, y_train, batch_size=128, epochs=20, verbose=0, callbacks=[LosswiseKerasCallback(tag='keras test')])
```

and you're good to go.
